### PR TITLE
Move some maths optimisations into a separate module

### PR DESCRIPTION
**Pull request by bmerry**
Apart from refactoring and making numba a hard requirement, add a ufunc to compute exp(2πix) more efficiently (and slightly more accurately).

### CODE REVIEW

File: katsdpimager/test/test_fast_math.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+class TestExpj2pi:
+    def setup(self):
+        gen = np.random.Generator(np.random.PCG64(42))
+        self.a = gen.random(100) * 1000.0 - 500.0
+        self.expected = np.exp(2j * np.pi * self.a)
+
+    def test_basic(self):
+        actual = fast_math.expj2pi(self.a)
+        np.testing.assert_allclose(self.expected, actual, rtol=1e-12)
```
Comment (ludwigschwardt) [inline katsdpimager/test/test_fast_math.py:17]: The signature of `assert_allclose` has the two inputs the other way around, which might help understanding any failures:
```Python
def assert_allclose(actual, desired, rtol=1e-7, atol=0, equal_nan=True, err_msg='', verbose=True):
```

File: katsdpimager/fast_math.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+@numba.vectorize([numba.complex64(numba.float32),
+                  numba.complex128(numba.float64)])
+def expj2pi(x):
+    """Equivalent to ``expj(2 * np.pi * x)`` where `x` is real.
+
+    x is reduced to a small value before multiplication, which
+    improves precision at a small cost in performance.
+    """
+    y = 2 * np.pi * (x - np.rint(x))
```
Comment (ludwigschwardt) [inline katsdpimager/fast_math.py:15]: Nice to know this trick.

File: katsdpimager/fast_math.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+"""Variants of standard maths functions written for speed."""
+
+import numpy as np
+import numba
+
+
+@numba.vectorize([numba.complex64(numba.float32),
```
Comment (ludwigschwardt) [inline katsdpimager/fast_math.py:7]: What is the speedup of Numba vs the original?

**Review by ludwigschwardt — CHANGES_REQUESTED** [review]

File: katsdpimager/test/test_fast_math.py (hunk 1 above)
Comment (bmerry) [reply katsdpimager/test/test_fast_math.py:17]: Thanks, will fix.

File: katsdpimager/fast_math.py (hunk 3 above)
Comment (bmerry) [reply katsdpimager/fast_math.py:7]: Looks like 40ns -> 25ns per element on my machine.

File: katsdpimager/test/test_fast_math.py (hunk 1 above)
Comment (bmerry) [reply katsdpimager/test/test_fast_math.py]: Done.

**Review by ludwigschwardt — APPROVED** [review]

**Pull request merged**